libraries/chain/controller.cpp

```diff
@@ -654,13 +655,17 @@ struct controller_impl {
    transaction_trace_ptr push_transaction( const transaction_metadata_ptr& trx,
                                            fc::time_point deadline,
                                            bool implicit,
-                                           uint32_t billed_cpu_time_us  )
+                                           uint32_t billed_cpu_time_us,
+                                           bool subjective = true)
```

Comment:
it looks like the "subjective" flag is just meant to account for whether we are in a state where our own subjective opinions about transactions apply.   If you look at how the white/black lists are implemented you will see that we have chosen to signal this based on `controller::block_status` where `controller::block_status::incomplete` implies that we are speculating OR producing a block and all other statuses imply that we are not in a mode where we can apply subjective measures.  

You should see the pattern in other parts of the code which will keep this from needing to be threaded down as part of the functional parameters.